File: java/LinkedList/reverse.java
import java.util.LinkedList;
import java.util.Scanner;
import java.util.Collections;

public class LinkedListReverse {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        LinkedList<Integer> list = new LinkedList<>();

        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();

        System.out.println("Enter elements:");
        for (int i = 0; i < n; i++) list.add(sc.nextInt());

        System.out.println("Original LinkedList: " + list);

        Collections.reverse(list); // brutal shortcut, acceptable in interviews
        System.out.println("Reversed LinkedList: " + list);
    }
}

File: python/LinkedList/reverse.py
class LinkedList(LinkedList):
    def reverse(self):
        prev = None
        curr = self.head
        while curr:
            nxt = curr.next
            curr.next = prev
            prev = curr
            curr = nxt
        self.head = prev

# Driver
ll = LinkedList()
n = int(input("Enter number of elements: "))
for _ in range(n):
    ll.append(int(input()))

print("Original LinkedList:")
ll.display()

ll.reverse()
print("Reversed LinkedList:")
ll.display()
